crates/merkle-tree/src/contract.rs

```diff
@@ -78,19 +78,24 @@ impl<'tx> ContractsStorageTree<'tx> {
 /// Tree data is persisted by a sqlite table 'tree_global'.
 pub struct StorageCommitmentTree<'tx> {
     tree: MerkleTree<PedersenHash, 251>,
-    storage: GlobalStorage<'tx>,
+    storage: StorageTrieReader<'tx>,
 }
 
 impl<'tx> StorageCommitmentTree<'tx> {
-    pub fn load(transaction: &'tx Transaction<'tx>, root: StorageCommitment) -> Self {
+    pub fn load(
+        transaction: &'tx Transaction<'tx>,
+        root: StorageCommitment,
+    ) -> anyhow::Result<Self> {
         let tree = MerkleTree::new(root.0);
-        let storage = GlobalStorage::new(transaction);
+        let storage = transaction
+            .storage_trie_reader()
+            .context("Loading storage")?;
 
-        Self { tree, storage }
+        Ok(Self { tree, storage })
     }
 
-    pub fn get(&self, address: ContractAddress) -> anyhow::Result<Option<ContractStateHash>> {
-        let value = self.tree.get(&self.storage, address.view_bits())?;
+    pub fn get(&mut self, address: ContractAddress) -> anyhow::Result<Option<ContractStateHash>> {
```

Comment:
Yeah its a result of using an internal statement.. I suppose we could use a `RefCell` or something similar to hide it?

Another reason it might be required is if one creates a storage adapter with an internal hash map cache - then it too would require `mut`.

Reply:
I'll see if its applicable for the database statement as well. I do think having the prepared statement is nice, but maybe its not worth it.

Reply:
We've decided to keep it `&mut` as this is fine for our current codebase.

Alternatives were:
1. `RefCell<PreparedStatement>` which has `!Send` implications
2. Moving the statement into `get` which would still work somewhat okay since rusqlite statement cache includes the statement's sql string.

Reply:
I'm open to changing this back and just relying on the sqlite statement caching. This is easily the thing that annoys me most.